### PR TITLE
Support raw gist URLs for loading

### LINKED ISSUE #470: Support loading raw gist urls
Like https://gist.githubusercontent.com/domoritz/3ab075a60cf92b8c19eb8a312e7b08b3/raw/ad920266b18cc7885b5c1b9a809907882084543a/spec

### PR DESCRIPTION
**Pull request by algomaster99**
Fixes #470

### CODE REVIEW

File: src/components/header/gist-modal/renderer.tsx [hunk 1]
```diff
@@ -140,7 +140,20 @@ class GistModal extends React.PureComponent<Props, State> {
       gistLoadClicked: true,
     });
 
-    const gistUrl = new URL(url, 'https://gist.github.com');
+    let gistUrl;
```
Comment (domoritz) [inline src/components/header/gist-modal/renderer.tsx]: Can you add a comment that explains at a high-level what is happening why?

**Comment by algomaster99** [comment]
@domoritz 
I have nested two conditions inside `onSelectGist`.
- `url.match(/gist.githubusercontent.com/)`
This will match the raw URL and accordingly extract `gistId`, `revision` and `filename` and set it in state. This is why you will see the fields get filled before the gist gets loaded. 
- `url.match(/gist.github.com/)`
This only extracts the `gistId` and load gists accordingly.

**Comment by domoritz** [comment]
Add comments to the code. For my future self ;-)

**Comment by algomaster99** [comment]
@domoritz If I forget, you can always ask me for an explanation. :)

**Comment by domoritz** [comment]
That's very kind of you. I will probably need to ask at some point.

**Pull request merged**